### ISSUE TITLE
teleport-connect 11.1.0 rpm package conflicts with slack 4.29.149 rpm package

### ISSUE THREAD

**Issue by cfeller** [gravitational/teleport#18859]
The teleport-connect rpm package (version 11.1.0 ) is unable to be installed alongside the slack rpm:

When attempting to install (via `dnf`), the following error is seen:

```
~> dnf install /tmp/teleport-connect-11.1.0.x86_64.rpm
Last metadata expiration check: 0:04:37 ago on Tue 29 Nov 2022 06:31:02 AM PST.
Dependencies resolved.
============================================================================================================================================
 Package                               Architecture                Version                          Repository                         Size
============================================================================================================================================
Installing:
 teleport-connect                      x86_64                      11.1.0-1                         @commandline                       84 M

Transaction Summary
============================================================================================================================================
Install  1 Package

Total size: 84 M
Installed size: 322 M
Is this ok [y/N]: y
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
Error: Transaction test error:
  file /usr/lib/.build-id/0d/9fad3f21b1e381cc66b5efcae8f3910ae0edc1 from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64
  file /usr/lib/.build-id/76/9d0269491ca2313fcd15d63059bac67bd8b1b3 from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64
  file /usr/lib/.build-id/c2/dcac7cb6a727685d10d4161a0d14988f12a0fa from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64
  file /usr/lib/.build-id/cb/dc9db5fec0e6129bddcb0c282be98be4c6966e from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64
```

Expected behavior:
teleport-connect installs

Current behavior:
teleport-connect fails to install due to a package conflict with another package

Bug details:
- Teleport version
  - `11.1.0` (rpm version `11.1.0-1.x86_64`)
- Recreation steps
  - Attempt to install teleport-connect version 11.1.0 on a system with the slack rpm already installed
- Debug logs
```
Error: Transaction test error:
  file /usr/lib/.build-id/0d/9fad3f21b1e381cc66b5efcae8f3910ae0edc1 from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64
  file /usr/lib/.build-id/76/9d0269491ca2313fcd15d63059bac67bd8b1b3 from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64
  file /usr/lib/.build-id/c2/dcac7cb6a727685d10d4161a0d14988f12a0fa from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64
  file /usr/lib/.build-id/cb/dc9db5fec0e6129bddcb0c282be98be4c6966e from install of teleport-connect-11.1.0-1.x86_64 conflicts with file from package slack-4.29.149-0.1.el8.x86_64```

**Comment by cfeller** [comment]
Notes: Previous versions of teleport-connect did not conflict with the slack rpm. Downgrading the slack rpm to version 4.28.184 also removes the conflict. Both of these packages (teleport-connect and slack) have seen updates past 24 hours so it is unclear who is doing the trampling here.
To be clear:
- teleport-connect 11.1.0 works with slack 4.28
- teleport-connect 11.0.3 works with slack 4.29
- teleport-connect 11.0.3 works with slack 4.28
- teleport-connect 11.1.0 **conflicts** with slack 4.29

**Comment by ravicious** [comment]
Thanks for a very detailed report.

It seems like it's a fairly common issue with apps that use electron-builder. Based on the "Additional information" section from https://github.com/ferdium/ferdium-app/issues/416, it looks like we'll have to pass some additional flags to fpm during our build process. It's interesting though that it wasn't an issue before and that the version of Slack also plays a role in here.

**Comment by ravicious** [comment]
This will be fixed once gravitational/webapps#1407 is backported to v11 and the next version is released, most likely 11.1.2.

In the meantime, you can use the 11.1.2-dev.ravicious.1 version I made to verify that the fix works.

* https://cdn.teleport.dev/teleport-connect-11.1.2-dev.ravicious.1.x86_64.rpm
* https://github.com/gravitational/webapps/compare/v11.1.1...v11.1.2-dev.ravicious.1
* https://github.com/gravitational/teleport/compare/v11.1.1...v11.1.2-dev.ravicious.1

**Comment by ravicious** [comment]
The fix has landed in 11.1.2. Unfortunately, while it fixes the original issue, I didn't realize that it introduced a regression – the symlink to `teleport-connect` doesn't get created.

**Comment by ravicious** [comment]
It seems that it's not the fault of that fix after all. I uninstalled Slack and installed Connect 11.1.1 – it also doesn't symlink the binary. 11.0.1 does symlink it to `/usr/bin`.

It might be a regression introduced in electron-builder v24, which we started using in 11.1.0.

**Comment by ravicious** [comment]
The missing `teleport-connect` symlink was fixed in gravitational/webapps#1434 and will be available in v11.1.4 or v11.2.0, whichever version is released first.
